env_setup/pre_setup.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+# WARNING:
+# This script will change python3 and pip3 as default python and pip.
+
+# tput codes
+RST="$(tput sgr0)"
+COL_BLACK="$(tput setaf 0)"
+COL_RED="$(tput setaf 1)"
+COL_GREEN="$(tput setaf 2)"
+COL_YELLOW="$(tput setaf 3)"
+COL_BLUE="$(tput setaf 4)"
+COL_WHITE="$(tput setaf 7)"
+TXT_UNDERLINE="$(tput smul)"
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    echo -e "$TXT_UNDERLINE---> MacOS detected so moving further with env_setup script. <---$RST\n"
+else
+    # exiting for now but later it can be updated
+    echo "Currently, script only works with MacOS!!!!!"
+    exit 1
+fi
+
+
+generate_err_and_exit() {
+    echo "$1" >&2
+    echo -e "!!! Exited with the Error Code !!!\n\n" >&2
+    kill 0
+}
+
+if_pckg_installed() {
+    pckg=$1
+    is_insatlled=`which $pckg`
+    [ -z "$is_insatlled" ] && echo "missing $pckg"
+}
+
+package_check() {
+    pckg=$1
+    local ret=$(if_pckg_installed $pckg)
+    if [[ "$ret" == "missing $pckg" ]]; then
+        generate_err_and_exit "$COL_RED\"$pckg\"$RST is not installed"
+    else
+        echo "$(which $pckg)"
+    fi
+}
+
+install_using_pip() {
+    pip install $1 >&2 > /dev/null
+}
+
+seupt_pip3_as_default() {
```

Comment:
Just want to make sure `seupt` is not the typo and we add it purposely.

Reply:
typos for sure. Updated.